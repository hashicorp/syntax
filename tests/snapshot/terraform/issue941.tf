instance_size = (                   #Comment
  length(var.instance_size) > 0 ? ( #Comment
    var.instance_size               #If instance size is provided, use it.
  )                                 #Comment
  :                                 #Comment
  (var.insane_mode ?
    lookup(local.insane_mode_instance_size_map, local.cloud, null) #If instance size is not provided and var.insane_mode is true, lookup in this table.
    :                                                              #
    lookup(local.instance_size_map, local.cloud, null)             #If instance size is not provided and var.insane_mode is false, lookup in this table.
  )
)

# Comment

variable "test" {
  default = "test"
}
