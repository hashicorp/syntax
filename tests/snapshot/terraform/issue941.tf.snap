>instance_size = (                   #Comment
#^^^^^^^^^^^^^ source.hcl.terraform variable.declaration.hcl variable.other.readwrite.hcl
#             ^ source.hcl.terraform variable.declaration.hcl
#              ^ source.hcl.terraform variable.declaration.hcl keyword.operator.assignment.hcl
#               ^ source.hcl.terraform variable.declaration.hcl
#                ^ source.hcl.terraform punctuation.section.parens.begin.hcl
#                 ^^^^^^^^^^^^^^^^^^^ source.hcl.terraform
#                                    ^ source.hcl.terraform comment.line.number-sign.hcl punctuation.definition.comment.hcl
#                                     ^^^^^^^ source.hcl.terraform comment.line.number-sign.hcl
>  length(var.instance_size) > 0 ? ( #Comment
#^^ source.hcl.terraform
#  ^^^^^^ source.hcl.terraform meta.function-call.hcl support.function.builtin.terraform
#        ^ source.hcl.terraform meta.function-call.hcl punctuation.section.parens.begin.hcl
#         ^^^ source.hcl.terraform meta.function-call.hcl support.constant.terraform
#            ^ source.hcl.terraform meta.function-call.hcl keyword.operator.accessor.hcl
#             ^^^^^^^^^^^^^ source.hcl.terraform meta.function-call.hcl variable.other.member.hcl
#                          ^ source.hcl.terraform meta.function-call.hcl punctuation.section.parens.end.hcl
#                           ^ source.hcl.terraform
#                            ^ source.hcl.terraform keyword.operator.hcl
#                             ^ source.hcl.terraform
#                              ^ source.hcl.terraform constant.numeric.integer.hcl
#                               ^ source.hcl.terraform
#                                ^ source.hcl.terraform keyword.operator.hcl
#                                 ^ source.hcl.terraform
#                                  ^ source.hcl.terraform punctuation.section.parens.begin.hcl
#                                   ^ source.hcl.terraform
#                                    ^ source.hcl.terraform comment.line.number-sign.hcl punctuation.definition.comment.hcl
#                                     ^^^^^^^ source.hcl.terraform comment.line.number-sign.hcl
>    var.instance_size               #If instance size is provided, use it.
#^^^^ source.hcl.terraform
#    ^^^ source.hcl.terraform support.constant.terraform
#       ^ source.hcl.terraform keyword.operator.accessor.hcl
#        ^^^^^^^^^^^^^ source.hcl.terraform variable.other.member.hcl
#                     ^^^^^^^^^^^^^^^ source.hcl.terraform
#                                    ^ source.hcl.terraform comment.line.number-sign.hcl punctuation.definition.comment.hcl
#                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl.terraform comment.line.number-sign.hcl
>  )                                 #Comment
#^^ source.hcl.terraform
#  ^ source.hcl.terraform punctuation.section.parens.end.hcl
#   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl.terraform
#                                    ^ source.hcl.terraform comment.line.number-sign.hcl punctuation.definition.comment.hcl
#                                     ^^^^^^^ source.hcl.terraform comment.line.number-sign.hcl
>  :                                 #Comment
#^^ source.hcl.terraform
#  ^ source.hcl.terraform
#   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl.terraform
#                                    ^ source.hcl.terraform comment.line.number-sign.hcl punctuation.definition.comment.hcl
#                                     ^^^^^^^ source.hcl.terraform comment.line.number-sign.hcl
>  (var.insane_mode ?
#^^ source.hcl.terraform
#  ^ source.hcl.terraform punctuation.section.parens.begin.hcl
#   ^^^ source.hcl.terraform support.constant.terraform
#      ^ source.hcl.terraform keyword.operator.accessor.hcl
#       ^^^^^^^^^^^ source.hcl.terraform variable.other.member.hcl
#                  ^ source.hcl.terraform
#                   ^ source.hcl.terraform keyword.operator.hcl
>    lookup(local.insane_mode_instance_size_map, local.cloud, null) #If instance size is not provided and var.insane_mode is true, lookup in this table.
#^^^^ source.hcl.terraform
#    ^^^^^^ source.hcl.terraform meta.function-call.hcl support.function.builtin.terraform
#          ^ source.hcl.terraform meta.function-call.hcl punctuation.section.parens.begin.hcl
#           ^^^^^ source.hcl.terraform meta.function-call.hcl support.constant.terraform
#                ^ source.hcl.terraform meta.function-call.hcl keyword.operator.accessor.hcl
#                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl.terraform meta.function-call.hcl variable.other.member.hcl
#                                              ^ source.hcl.terraform meta.function-call.hcl punctuation.separator.hcl
#                                               ^ source.hcl.terraform meta.function-call.hcl
#                                                ^^^^^ source.hcl.terraform meta.function-call.hcl support.constant.terraform
#                                                     ^ source.hcl.terraform meta.function-call.hcl keyword.operator.accessor.hcl
#                                                      ^^^^^ source.hcl.terraform meta.function-call.hcl variable.other.member.hcl
#                                                           ^ source.hcl.terraform meta.function-call.hcl punctuation.separator.hcl
#                                                            ^ source.hcl.terraform meta.function-call.hcl
#                                                             ^^^^ source.hcl.terraform meta.function-call.hcl constant.language.hcl
#                                                                 ^ source.hcl.terraform meta.function-call.hcl punctuation.section.parens.end.hcl
#                                                                  ^ source.hcl.terraform
#                                                                   ^ source.hcl.terraform comment.line.number-sign.hcl punctuation.definition.comment.hcl
#                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl.terraform comment.line.number-sign.hcl
>    :                                                              #
#^^^^ source.hcl.terraform
#    ^ source.hcl.terraform
#     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl.terraform
#                                                                   ^ source.hcl.terraform comment.line.number-sign.hcl punctuation.definition.comment.hcl
>    lookup(local.instance_size_map, local.cloud, null)             #If instance size is not provided and var.insane_mode is false, lookup in this table.
#^^^^ source.hcl.terraform
#    ^^^^^^ source.hcl.terraform meta.function-call.hcl support.function.builtin.terraform
#          ^ source.hcl.terraform meta.function-call.hcl punctuation.section.parens.begin.hcl
#           ^^^^^ source.hcl.terraform meta.function-call.hcl support.constant.terraform
#                ^ source.hcl.terraform meta.function-call.hcl keyword.operator.accessor.hcl
#                 ^^^^^^^^^^^^^^^^^ source.hcl.terraform meta.function-call.hcl variable.other.member.hcl
#                                  ^ source.hcl.terraform meta.function-call.hcl punctuation.separator.hcl
#                                   ^ source.hcl.terraform meta.function-call.hcl
#                                    ^^^^^ source.hcl.terraform meta.function-call.hcl support.constant.terraform
#                                         ^ source.hcl.terraform meta.function-call.hcl keyword.operator.accessor.hcl
#                                          ^^^^^ source.hcl.terraform meta.function-call.hcl variable.other.member.hcl
#                                               ^ source.hcl.terraform meta.function-call.hcl punctuation.separator.hcl
#                                                ^ source.hcl.terraform meta.function-call.hcl
#                                                 ^^^^ source.hcl.terraform meta.function-call.hcl constant.language.hcl
#                                                     ^ source.hcl.terraform meta.function-call.hcl punctuation.section.parens.end.hcl
#                                                      ^^^^^^^^^^^^^ source.hcl.terraform
#                                                                   ^ source.hcl.terraform comment.line.number-sign.hcl punctuation.definition.comment.hcl
#                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl.terraform comment.line.number-sign.hcl
>  )
#^^ source.hcl.terraform
#  ^ source.hcl.terraform punctuation.section.parens.end.hcl
>)
#^ source.hcl.terraform punctuation.section.parens.end.hcl
>
># Comment
#^ source.hcl.terraform comment.line.number-sign.hcl punctuation.definition.comment.hcl
# ^^^^^^^^ source.hcl.terraform comment.line.number-sign.hcl
>
>variable "test" {
#^^^^^^^^ source.hcl.terraform meta.block.hcl entity.name.type.terraform
#        ^ source.hcl.terraform meta.block.hcl
#         ^^^^^^ source.hcl.terraform meta.block.hcl variable.other.enummember.hcl
#               ^ source.hcl.terraform meta.block.hcl
#                ^ source.hcl.terraform meta.block.hcl punctuation.section.block.begin.hcl
>  default = "test"
#^^ source.hcl.terraform meta.block.hcl
#  ^^^^^^^ source.hcl.terraform meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#         ^ source.hcl.terraform meta.block.hcl variable.declaration.hcl
#          ^ source.hcl.terraform meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#           ^ source.hcl.terraform meta.block.hcl variable.declaration.hcl
#            ^ source.hcl.terraform meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#             ^^^^ source.hcl.terraform meta.block.hcl string.quoted.double.hcl
#                 ^ source.hcl.terraform meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>}
#^ source.hcl.terraform meta.block.hcl punctuation.section.block.end.hcl
>